tests/src/tracing/regress/GitHub_22337/GitHub_22337.cs

```diff
@@ -0,0 +1,190 @@
+                return false;
+            }
+
+            m_setProfilerSamplingRateMethod = configurationType.GetMethod(
+                "SetProfilerSamplingRate",
+                BindingFlags.NonPublic | BindingFlags.Instance);
+            if (m_setProfilerSamplingRateMethod == null)
+            {
+                Console.WriteLine("setProfilerSamplingRate == null");
+                return false;
+            }
+
+            return true;
+        }
+    }
+
+    class Program
+    {
+        private static MethodInfo m_enableMethod;
+        private static MethodInfo m_disableMethod;
+
+        public static void Enable(TraceConfiguration traceConfig)
+        {
+            m_enableMethod.Invoke(
+                null,
+                new object[]
+                {
+                    traceConfig.ConfigurationObject
+                });
+        }
+
+        public static void Disable()
+        {
+            m_disableMethod.Invoke(
+                null,
+                null);
+        }
+
+        public static void GitHub_22247(TraceConfiguration config)
+        {
+            // Enable tracing.
+            Enable(config);
+            Disable();
+        }
+
+        static int Main(string[] args)
+        {
+            TimeSpan profSampleDelay = TimeSpan.FromMilliseconds(1);
+            string outputFile = "default.netperf";
+
+            Assembly SPC = typeof(System.Diagnostics.Tracing.EventSource).Assembly;
+            Type eventPipeType = SPC.GetType("System.Diagnostics.Tracing.EventPipe");
+            m_enableMethod = eventPipeType.GetMethod("Enable", BindingFlags.NonPublic | BindingFlags.Static);
+            m_disableMethod = eventPipeType.GetMethod("Disable", BindingFlags.NonPublic | BindingFlags.Static);
+
+            // Setup the configuration values.
+            uint circularBufferMB = 1024; // 1 GB
+            uint level = 5; // Verbose
+
+            // Create a new instance of EventPipeConfiguration.
+            TraceConfiguration config = new TraceConfiguration(outputFile, circularBufferMB);
+            // Setup the provider values.
+            // Public provider.
+            string providerName = "Microsoft-Windows-DotNETRuntime";
+            UInt64 keywords = 0x4c14fccbd;
+
+            // Enable the provider.
+            config.EnableProvider(providerName, keywords, level);
+
+            // Private provider.
+            providerName = "Microsoft-Windows-DotNETRuntimePrivate";
+            keywords = 0x4002000b;
+
+            // Enable the provider.
+            config.EnableProvider(providerName, keywords, level);
+
+            // Sample profiler.
+            providerName = "Microsoft-DotNETCore-SampleProfiler";
+            keywords = 0x0;
+
+            // Enable the provider.
+            config.EnableProvider(providerName, keywords, level);
+
+            // Set the sampling rate.
+            config.SetSamplingRate(profSampleDelay);
+
+            for (int i = 0; i < 1_000_000; ++i)
```

Comment:
Rather than using this test we should probably use a test that hits this condition - one that enables `System.Runtime` and then disables it before m_timer gets initialized.

Reply:
Also I don't think this test is enabling the right EventSource - It should be enabling/disabling `System.Runtime` instead of `Microsoft-Windows-DotNETRuntime`/`Microsoft-Windows-DotNETRuntimePrivate` to hit the codepath this PR changes.

Reply:
Agreed, adding this test might cause timeouts on CI runs because the original issue deadlocks.